plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -857,9 +857,9 @@ class FossId internal constructor(
         }
 
         val matchedLines = mutableMapOf<Int, MatchedLines>()
-        val snippets = runBlocking(Dispatchers.IO) {
-            pendingFiles.map {
-                async {
+        val snippets = pendingFiles.associateWith {
+            val filteredSnippets = lazy {
+                runBlocking(Dispatchers.IO) {
```

Comment:
Since `getRawResults()` is a *suspend* function, it should not call `runBlocking()`. Use `withContext(Dispatchers.IO)` instead.

Reply:
Thanks for the hint. Unfortunately, `withContext` is a suspending function and it cannot be used with `lazy`, because the latter does not support suspension.

Reply:
To avoid `runBlocking` altogether, I refactored the code to generate a `Sequence` as @sschuberth suggested.
Unfortunately, the function building a `Sequence` cannot be suspendable.
Therefore, I went with a `Flow`: values are emitted asynchronously and if the limit has been reached, the collecting job is cancelled.